contracts/exchange/EndemicExchangeCore.sol

```diff
@@ -137,6 +242,70 @@ abstract contract EndemicExchangeCore {
         }
     }
 
+    function _requireSupportedErc20Token(address paymentErc20TokenAddress)
+        internal
+        view
+    {
+        if (paymentErc20TokenAddress == ZERO_ADDRESS) return;
+
+        require(
+            supportedErc20Addresses[paymentErc20TokenAddress],
+            "ERC20 Token is not supported for paying on Endemic!"
+        );
+    }
+
+    function _requireCorrectValueProvided(
+        uint256 requiredValue,
+        address paymentErc20TokenAddress,
+        address buyer
+    ) internal view {
+        if (paymentErc20TokenAddress == ZERO_ADDRESS) {
+            _requireCorrectEtherValueProvided(requiredValue);
+        } else {
+            _requireCorrectErc20ValueProvided(
+                requiredValue,
+                paymentErc20TokenAddress,
+                buyer
+            );
+        }
+    }
+
+    function _requireCorrectEtherValueProvided(uint256 requiredValue)
+        internal
+        view
+    {
+        if (msg.value < requiredValue) {
+            revert InvalidValueProvided();
+        }
+    }
+
+    function _requireCorrectErc20ValueProvided(
+        uint256 requiredValue,
+        address paymentErc20TokenAddress,
+        address buyer
+    ) internal view {
+        IERC20 ERC20PaymentToken = IERC20(paymentErc20TokenAddress);
+
+        uint256 contractAllowance = ERC20PaymentToken.allowance(
+            buyer,
+            address(this)
+        );
+
+        if (contractAllowance < requiredValue) {
+            revert InvalidValueProvided();
+        }
+    }
+
+    function _updateSupportedErc20Tokens(address _erc20TokenAddressToSupport)
+        internal
+    {
+        if (_erc20TokenAddressToSupport == ZERO_ADDRESS) {
+            revert InvalidAddress();
+        }
+
+        supportedErc20Addresses[_erc20TokenAddressToSupport] = true;
```

Comment:
Sta kazes da ova metoda prima param `bool isEnabled` tako da u nekom trenutku mozemo i maknit token ako bude za to potrebe?